src/bam/simulate_reads.rs

```diff
@@ -0,0 +1,181 @@
+use anyhow::Result;
+use bio::io::fasta;
+use rand::prelude::ThreadRng;
+use rand::Rng;
+use rust_htslib::bam;
+use rust_htslib::bam::Read;
+use std::collections::HashMap;
+use std::path::Path;
+use uuid::Uuid;
+
+pub fn simulate_reads<P: AsRef<Path>>(
+    bam: P,
+    input_ref: P,
+    output_bam: P,
+    output_ref: P,
+    chr: String,
+    start: u64,
+    end: u64,
+    pairs: bool,
+) -> Result<()> {
+    let mut fasta_reader = fasta::IndexedReader::from_file(&input_ref)?;
+    fasta_reader.fetch(&chr, start - 1, end - 1)?;
+    let mut reference = Vec::new();
+    fasta_reader.read(&mut reference)?;
+    let mut rng = rand::thread_rng();
+    let alphabet = [b'A', b'C', b'G', b'T'];
+
+    //Build artificial reference
+    let mut artificial_reference = Vec::new();
+    add_random_bases(end - start, &mut artificial_reference, &mut rng, &alphabet)?;
+    let mut altered_bases = init_altered_bases(&reference, &artificial_reference)?;
+    let mut fa_writer = fasta::Writer::to_file(output_ref)?;
+    let ref_id = Uuid::new_v4().to_hyphenated().to_string();
+    fa_writer.write(&ref_id, None, &artificial_reference)?;
+
+    let mut bam_reader = bam::IndexedReader::from_path(bam)?;
+    bam_reader.fetch((chr.as_bytes(), start - 1, end))?;
```

Comment:
Instead of converting to zero based ranges here and below (start, end), I'd rather expect this function to receive a zero based range and convert it before in the CLI.